src/utils/viewer/viewer.cpp

```diff
@@ -352,14 +373,27 @@ void Viewer::drawEvent() {
     physicsManager_->debugDraw(projM * camM);
   }
 
+  if (displayPerformanceInfo) {
+    Mn::Color3 textColor{1.0f};
+    textRenderer_->draw(textColor);
+  }
+
   swapBuffers();
-  timeline_.nextFrame();
   redraw();
+
+  totalFrames_++;
+  if (displayPerformanceInfo && (totalFrames_ % fpsUpdateFrequency) == 0) {
+    updateText(totalFrames_ / timeline_.previousFrameTime());
```

Comment:
:thinking: so ... let's say you have 1500 `totalFrames_` and previous frame time was 0.016 seconds. That'll give you a FPS of 93750. Or am I missing something?

Also, what about some moving average? That's what I originally meant when I said "update every 15 frames".

Reply:
You mixed ```previousFrameTime()``` and ```previousFrameDuration()```. 
```
previousFrameTime 
Time at previous frame (in seconds)
Returns time elapsed since start() was called. If the timeline is stopped, the function returns 0.0f.
```
So here it gives you the average FPS ever since the simulator runs, which is not what I meant.
Good catch. I need to change it.